### ISSUE TITLE
Small Pipes replacement...

### ISSUE THREAD

**Issue by datahead8888** [Secretchronicles/TSC#308]
This is a task to create a replacement for the small pipes.
I noticed no open task currently corresponds to them, so I figured I'd open one for tracking.
I understand the goal is to get these in release-2.0.0.  If not, we can always remove the marker.

@sydneyjd's draft for small pipes is at: http://forum.secretchronicles.de/topics/158?page=1
--toward the bottom

**Comment by Quintus** [comment]
The small pipes are so rarely used that I don’t think not having them in 2.0.0 will hurt us.

Vale,
Quintus

**Comment by datahead8888** [comment]
They are in world 1, but you are correct they are not prominent.  I had thought the goal was to put them in release 2.0, but we probably need to reevaluate what's critical.  It's probably just a good-to-have-if-we-have-time item.

**Comment by datahead8888** [comment]
@sydneyjd, @Quintus, @sauer2, I took the small pipes jpg from http://forum.secretchronicles.de/topics/158?page=1 and converted it to the necessary PNG files for small pipes.  I then took a look at it in the "archepipical example" (Now I'm inventing new words).

![small_pipes_in_a_level](https://cloud.githubusercontent.com/assets/6478551/5605973/cef337a0-93e6-11e4-93f1-0f8f8ee42304.png)

Conclusions: I was probably wrong about the horizontal lines suggestion I made earlier.  Since the pipe is in two pieces, it makes more sense to have one with lines and one with another.  Also, many pieces of small pipe without lines can be jerry rigged together as is done here.

We may need to do some more checking on boundaries by testing in some levels.  I had thought the old small pipes had smaller pipe shapes (WITHIN the set boundaries).  Also, we'll need an SVG for the drafts directory.

**Comment by datahead8888** [comment]
Yep, we've got a sizing issue.  Here's a screenshot.  If you take a look, you'll see Alex is floating a bit above the small pipe top.

![boundary_problem](https://cloud.githubusercontent.com/assets/6478551/5606041/631e7d6c-93e8-11e4-90a5-cc7a5d5f1e4d.png)

For the body, though, no such problem exists:
![good_bondary](https://cloud.githubusercontent.com/assets/6478551/5606042/740a6956-93e8-11e4-965e-7c767e13769f.png)

I'd be really hesitant to try to squeeze down the boundaries by changing the image settings after seeing the problems we ran into with enemy placement for the large pipes.  I realize small pipes are much smaller and don't usually have Flyons coming out of them, but if it's possible to generate an SVG that's a carbon copy of the original dimensions, we will have a guarantee these problems never occur.  I'm open to other opinions on this, though.

**Comment by sauer2** [comment]
Nah, you're right. The ingame size should never ever be changed. Changed 
SVGs and their generated PNGs probably cause a lot less work.

**Comment by sydneyjd** [comment]
@datahead8888 Im working on it...

**Comment by datahead8888** [comment]
I wouldn't say "never" :) I would say we should be very cautious about doing it in release-2.0.0 this late and that we should be weary of it in the 2.X release series.  In version 3.0, anything will be fair game, though.

**Comment by datahead8888** [comment]
@sydneyjd -- the new dimensions look good:

![pipes_in_action](https://cloud.githubusercontent.com/assets/6478551/5609452/ebd3a502-946e-11e4-8ab2-a3a1941cc86c.png)

...and those were configured as green pipes...changed through...compatibility code...

**Comment by datahead8888** [comment]
Small pipes are added to the release-2.0.0 branch.  Here's a screenshot of the "archepipical" example, which demonstrates a compatibility conversion of small green pipes to small grey pipes:
![small_pipes_testing_release](https://cloud.githubusercontent.com/assets/6478551/5611046/e4f62046-948e-11e4-8699-23584c6a7fc3.png)

**Comment by sydneyjd** [comment]
Cool, i will test them soon.

**Comment by datahead8888** [comment]
> And when it is, we’ll forget about it, because it’s not autoclosing

The commit had the closes syntax, so I think it would have autoclosed once this was in devel.  I don't have a problem with manually closing it, though.

**Comment by Quintus** [comment]
> The commit had the closes syntax, so I think it would have autoclosed once this was in devel.

Somehow I overlooked that, sorry.

Vale,
Quintus
